nix/fetchNixpkgs.nix

```diff
@@ -0,0 +1,56 @@
+{ rev                             # The Git revision of nixpkgs to fetch
+, sha256                          # The SHA256 of the downloaded data
+, outputSha256 ? null             # The SHA256 output hash
+, system ? builtins.currentSystem # This is overridable if necessary
+}:
+
+with {
```

Comment:
I shamelessly cargo-culted it [from here](https://nixos.wiki/wiki/How_to_fetch_Nixpkgs_with_an_empty_NIX_PATH) and added the fixed-output hash behavior.

You're also correct, we can use a normal if conditional, I'll change that since I agree it's straight-forward that way.